### PR TITLE
RR-612 Timeline prison movements nunjucks template

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR modifies the nunjucks template for the Timeline, so that a prisoner's history of prisons is included in the view as follows:

image

As can be seen, the formatting for the prison movements is different to other events and does not include the name of the user who performed an action.

I will add an integration test in my next PR and also enable the `INCLUDE_PRISON_TIMELINE_EVENTS_ENABLED` feature toggle.

### CODE REVIEW

File: server/views/pages/overview/partials/timelineTab/timelineTabContents.njk [hunk 1]
```diff
@@ -7,30 +7,41 @@
...
+              </div>
+
+              <p class="moj-timeline__date">
+                <time datetime="{{ event.timestamp }}">{{ event.timestamp | formatDate('D MMMM YYYY') }}</time>
               </p>
-            </div>
-            {% endif %}
 
+          {% else %}
+            <div class="moj-timeline__item" data-qa-event-type="{{ event.eventType }}">
```
Comment (mattwills-moj-digital) [inline server/views/pages/overview/partials/timelineTab/timelineTabContents.njk:21]: The remainder of the template is the same as before...

File: server/views/pages/overview/partials/timelineTab/timelineTabContents.njk [hunk 2]
```diff
@@ -7,30 +7,41 @@
...
-            <div class="moj-timeline__description govuk-!-display-none-print">
-              <p class="govuk-body govuk-!-margin-bottom-0">
-                <a href="/plan/{{ prisonerSummary.prisonNumber }}/view/overview">
-                  {% if event.eventType == 'ACTION_PLAN_CREATED' %}
-                    View {{ prisonerSummary.firstName }} {{ prisonerSummary.lastName | title }}'s learning and work progress
-                  {% elif event.eventType == 'GOAL_UPDATED' or event.eventType == 'GOAL_CREATED' or event.eventType == 'MULTIPLE_GOALS_CREATED' %}
-                    View <span class="govuk-visually-hidden">{{ prisonerSummary.firstName }} {{ prisonerSummary.lastName | title }}'s </span>goals
-                  {% endif %}
-                </a>
+          {% if event.eventType == 'PRISON_ADMISSION' or event.eventType == 'PRISON_TRANSFER' or event.eventType == 'PRISON_RELEASE' %}
```
Comment (mattwills-moj-digital) [inline server/views/pages/overview/partials/timelineTab/timelineTabContents.njk:10]: Due to the way prison movement events need to appear (e.g. without the `actionedByDisplayName`), we need a separate section to format them.

File: server/filters/formatPrisonMovementEventFilter.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import type { TimelineEvent } from 'viewModels'
+
+export default function formatPrisonMovementEventFilter(event: TimelineEvent): string {
```
Comment (mattwills-moj-digital) [inline server/filters/formatPrisonMovementEventFilter.ts:3]: This is similar to the existing `formatTimelineEvent` filter, except that the formatted value needs to contain the prison name(s).

File: server/services/timelineService.ts [hunk 4]
```diff
@@ -43,12 +43,15 @@ export default class TimelineService {
 
     // Lookup the prison for the first event by itself to prevent a race condition on the `PrisonService`
     // If the `PrisonStore` cache is empty/stale, attempting to lookup all the event's prisons in a `Promise.all`
-    // loop results in a race condition where all prison look ups run (essentially) at the same time, and they all try
+    // loop results in a race condition where all prison look-ups run (essentially) at the same time, and they all try
```
Comment (mattwills-moj-digital) [inline server/services/timelineService.ts:46]: IntelliJ was prompting me to fix this. I tried to ignore it, but failed! 😂

Comment (nathanrussell-moj-digital) [reply server/services/timelineService.ts:46]: 🤣

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍

**Pull request merged**